test/syntax.jl

```diff
@@ -1447,3 +1447,11 @@ f27129(x = 1) = (@Base._inline_meta; x)
 for meth in methods(f27129)
     @test ccall(:jl_uncompress_ast, Any, (Any, Any), meth, meth.source).inlineable
 end
+
+# issue #27710
+struct Foo27710{T} end
+function test27710()
+    types(::Foo27710{T}) where T = T
+    T = types(Foo27710{Int64}())
```

Comment:
This is a pretty inscrutable test. Is assigning this particular value to `T` significant or is it just the fact that something is being assigned to it that's important?

Reply:
Both. The inner function needs to work, and the assigned variable here needs to be called `T`.

Reply:
Right, but you could do something like this instead, right:
```jl
T = nothing
types(::Foo27710{T}) where T = T
return types(Foo27710{Int64}())
```

Reply:
Sure, or even better return `(T, types(Foo27710{Int64}()))`, to make sure the assignment works too. But it seems like a pretty minor point to me.

Reply:
I was just trying to understand it, not complain 😄